src/JuMP.jl

```diff
@@ -29,6 +29,9 @@ if VERSION < v"0.7-"
     const SparseArrays = Compat.SparseArrays
     const Printf = Compat.Printf
     const Sys = Compat.Sys
+    using Compat: rmul!
+    # because iteration of CartesianIndex is broken in 0.6
+    Base.Tuple(idx::CartesianIndex{N}) where {N} = tuple((idx[i] for i ∈ 1:N)...)
```

Comment:
This is type piracy. It's better to define a function with a new name rather than overloading base methods that weren't meant to be overloaded.

Also, where is this used? If it's for every `getindex` with `JuMPArray` the splatting could have serious performance implications.

Reply:
This is used in `KeyIterator` and in printing (do a search for `CartesianIndices`).  As I understand it the splatting only needs to compile once for each *rank* of array it is called on, so it shouldn't be to bad.  This is the replacement for `ind2sub`.

Ok, I was trying very hard to get rid of as much 0.6 code as possible as it'll probably be ancient history in a matter of weeks, but if you prefer I can define a new function instead.

Reply:
> Ok, I was trying very hard to get rid of as much 0.6 code as possible as it'll probably be ancient history in a matter of weeks, but if you prefer I can define a new function instead.

I'm reviewing this as code that we'll be responsible for supporting and responding to issues about. Any ugly hacks need a good justification, and in this case the cleaner version is easy to do.